dev-java/parsington/parsington-9999.ebuild

```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="doc source test"
+JAVA_TESTING_FRAMEWORKS="junit-jupiter"
+
+inherit java-pkg-2 java-pkg-simple
+
+DESCRIPTION="A mathematical expression parser for infix expression strings"
+HOMEPAGE="https://github.com/scijava/parsington"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/scijava/parsington.git"
+	S="${WORKDIR}/${P}"
+	MAVEN_ID="org.scijava:parsington:9999"
+else
+	SRC_URI="
+		https://github.com/scijava/${PN}/archive/refs/tags/${P}.tar.gz -> ${P}-sources.tar.gz
+	"
+	S="${WORKDIR}/${PN}-${P}"
+	MAVEN_ID="org.scijava:parsington:3.1.0"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="BSD-2"
+SLOT="0"
+
+BDEPEND="
+	>=virtual/jdk-1.8:*
+	test? (
+	      >=dev-java/junit-5.9.1
+	)
+"
+
+RDEPEND="
+	>=virtual/jre-1.8:*
+"
+
+JAVA_SRC_DIR="src/main/java"
+JAVA_MAIN_CLASS="org.scijava.parsington.Main"
+
+JAVA_TEST_GENTOO_CLASSPATH="junit-5.9.1"
```

Comment:
I'm getting this error: `!!! ERROR: Package junit-5.9.1 was not found!`

The version of junit I have installed is 5.9.2. I'm not super familiar with java packaging, but is there any way we can write this so it doesn't break every time junit updates. Does it work if you simply don't specify the version number?

Reply:
Interesting. 
I am not :100:  either :worried: 
But I just pushed it as "junit-5", as many ebuilds in the dev-java subdir of the gentoo repo appear to have a line like this:

`
JAVA_TEST_GENTOO_CLASSPATH="junit-4"
`
I hope it works. 
I didn't build with the test flag.

Also, I am not sure how your are testing, but I suggest 

`emerge -av1`

Since these are really being added as a dependency tree, and dependencies are better installed with the `1`, for depclean to operate graciously

Regards